src/app/ClusterObjectCommandSenderDelegate.h

```diff
@@ -0,0 +1,87 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <app/CommandSender.h>
+#include <app/common/StatusElement.h>
+#include <app/data-model/Decode.h>
+
+#include <function>
+
+namespace chip {
+namespace app {
+
+template <typename ClusterObjectT>
+class ClusterObjectCommandSenderDelegate final : public CommandSender::Delegate
+{
+public:
+    template <typename ClusterObjectT>
+    using OnResponseCallbackType<ClusterObjectT> =
+        std::function<void(const CommandSender *, const CommandPath::Type &, const StatusElement::Type &, ClusterObjectT &)>;
+
+    template <>
+    using OnResponseCallbackType<void> =
+        std::function<void(const CommandSender *, const CommandPath::Type &, const StatusElement::Type &)>;
+
+    using OnErrorCallbackTyoe = std::function<void(const CommandSender *, CHIP_ERROR aError)>;
+
+    using DeleterType = std::function<void(CommandSender *, ClusterObjectCommandSenderDelegate<ClusterObjectT>)>;
```

Comment:
The naming in the original delegate (and even, in the method below) is `ReleaseCommandSender`, but this has quite a different name.

We should be consistent and likely, rename this to `OnReleaseSenderType`.